### ISSUE TITLE
Automatic config Learning rate scheduler and batch normalization with momentum

### ISSUE THREAD

**Issue by MinWang1997** [PyTorchLightning/pytorch-lightning#10352]
## 🚀 Feature

 Easy way to config optimization:
Learning rate scheduler and batch normalization with momentum

### Motivation

I reorganized the source code of one repository to pytorch lighting version but I just noticed that they used Learning rate scheduler and batch normalization with momentum.

It seems that pytorch does not have batch normalization with momentum since I did not get result after searching it via google.

### Pitch

Make the manual way of optimization more easier or add these schedulers to automatic way.

### Additional context

Do you think these schedulers are important for the models ?
I have already run a baseline and try to do modifications and then run experiments without them. Should I keep all the experiments as before without schedulers or I need to add them to get better performance and fare comparison ? 

![E87E3652-231E-4663-8AC1-6834EB9459E2](https://user-images.githubusercontent.com/75472853/140326298-adf25ddb-c552-44d4-8692-b8cc4ca1d65b.jpeg)
![EAED5DA1-6254-4334-B90A-53D2904D0D99](https://user-images.githubusercontent.com/75472853/140326307-19f653d0-a9a8-4680-9f6e-4de3f70b3098.jpeg)

cc @borda

**Comment by awaelchli** [comment]
Hello, is it the BNMomentumScheduler from here?
https://github.com/facebookresearch/votenet/blob/main/train.py

If I see correctly, this one is just stepping every epoch. So you could put this step in the `on_epoch_start` or `on_epoch_end` hook in the LightningModule.

> Do you think these schedulers are important for the models ?

I'm not sure. I couldn't find it in the paper. If you share the exact method you are trying to convert, can have another look.

> Should I keep all the experiments as before without schedulers or I need to add them to get better performance and fare comparison ?

If plan to take this as a baseline and build on top of it, then yes, for a fair comparison you will have to keep them.

**Comment by MinWang1997** [comment]
Thank you for your reply.

They also use defined a trainer similar to pytorchlightning. It would be nice if you could add them to trainer class.
The learning rate scheduler and Batch normalization  momentum are initiated in the [trainer class.](https://github.com/ethnhe/FFB6D/blob/master/ffb6d/train_ycb.py)


The learning rate scheduler is from torch.optim.lr_scheduler

The BNM is defined in [pytorch_utils.py](https://github.com/ethnhe/FFB6D/blob/master/ffb6d/models/pytorch_utils.py)

**Comment by MinWang1997** [comment]
Hi,

As for the learning rate scheduler, 

Can I simply add learning scheduler name as this [example](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.lightning.html#pytorch_lightning.core.lightning.LightningModule.configure_optimizers) ?

def configure_optimizers(self):
    optimizer = Adam(...)
    return {
        "optimizer": optimizer,
        "lr_scheduler": {
            "scheduler": ReduceLROnPlateau(optimizer, ...),
            "monitor": "metric_to_track",
            "frequency": "indicates how often the metric is updated"
            # If "monitor" references validation metrics, then "frequency" should be set to a
            # multiple of "trainer.check_val_every_n_epoch".
        },
    }



How can we specify the parameters as torch does?
```

lr_scheduler = CyclicLR(
            optimizer, base_lr=1e-5, max_lr=1e-3,
            cycle_momentum=False,
            step_size_up=config.n_total_epoch * train_ds.minibatch_per_epoch // clr_div // args.gpus,
            step_size_down=config.n_total_epoch * train_ds.minibatch_per_epoch // clr_div // args.gpus,
            mode='triangular')
```


Best wishes,
Min

**Comment by tchaton** [comment]
Dear @MinWang1997 ,

I believe it could be possible to add support for a `bn_scheduler`, but I don't believe this is a common use case.

**Comment by MinWang1997** [comment]
> As for the learning rate scheduler,
> 
> Can I simply add learning scheduler name as this [example](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.lightning.html#pytorch_lightning.core.lightning.LightningModule.configure_optimizers) ?

Thank you for your reply!
Could you have a look at this question?
I just need to pass lr_scheduler by myself right?
Will lightning automatic optimization add one default scheduler?

**Comment by akihironitta** [comment]
@MinWang1997 For all your questions on simple learning rate schedulers, please have a look at my comment below:

> I just need to pass lr_scheduler by myself right?

Yes, you need to have a lr_scheduler with an optimizer in `configure_optimizers` if you want to use one.

> Will lightning automatic optimization add one default scheduler ... ?

No, Lightning doesn't add any lr_scheduler if you have `configure_optimizers` returning no lr scheduler.

> How can we specify the parameters as torch does?

You can specify parameters for your lr scheduler.

```python
# slightly modified version of examples in the docs
def configure_optimizers(self):
    optimizer = YourOptimizer(...)
    lr_scheduler = YourLRScheduler(optimizer, ...)
    return {"optimizer": optimizer, "lr_scheduler": lr_scheduler}

def configure_optimizers(self):
    optimizer = YourOptimizer(...)
    lr_scheduler = YourLRScheduler(optimizer, ...)
    lr_scheduler_config = {
        "scheduler": lr_scheduler,
        # "interval": "epoch",  # default
        # "frequency": 1,  # default
        # "monitor": "val_loss",  # default
        # "name": None,  # default
    }
    return {"optimizer": optimizer, "lr_scheduler": lr_scheduler_config}
```

Hope this helps. See the docs for more details: https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.lightning.html#pytorch_lightning.core.lightning.LightningModule.configure_optimizers

**Comment by MinWang1997** [comment]
> Dear @MinWang1997 ,
> 
> I believe it could be possible to add support for a `bn_scheduler`, but I don't believe this is a common use case.

Thank you.
Before your implementation, could you tell me how to add it by myself under lightning framework?
